client.go

```diff
@@ -554,23 +554,36 @@ func (client *client) update(data *MetadataResponse) ([]string, error) {
 
 	var err error
 	for _, topic := range data.Topics {
+
+		delete(client.metadata, topic.Name)
+		delete(client.cachedPartitionsResults, topic.Name)
+
 		switch topic.Err {
 		case ErrNoError:
 			break
-		case ErrLeaderNotAvailable, ErrUnknownTopicOrPartition:
+		case ErrInvalidTopic: // don't retry, don't store partial results
+			err = topic.Err
```

Comment:
Ah, I thought we checked the error return before the retry return, but we do not, so this is OK, good catch :+1: